### PR TITLE
Node: configurable `util.inspect()` options

### LINKED ISSUE #316: Format to print an object at infinite depth?
Is there a format that will print a given object without keys being equal `[Object]` (e.g. that doesn't care about depth?). I tried `debug('%o', obj)` but that only will print up to a certain depth.

### PR DESCRIPTION
**Pull request by TooTallNate**
Via env variables by default. So to get more object depth,
you pass the `DEBUG_DEPTH=10` env var.

For the `showHidden` option, you set `DEBUG_SHOW_HIDDEN=on`.

See the Node.js docs for the complete list of `util.inspect()` options:
https://nodejs.org/api/util.html#util_util_inspect_object_options

Fixes #316. Perhaps others.

### CODE REVIEW

**Comment by thebigredgeek** [comment]
@TooTallNate can you update docs?

**Comment by thebigredgeek** [comment]
@TooTallNate any update on docs man?  Also, we need to resolve conflicts on this.  I'd like to release 2.4 this week

**Comment by TooTallNate** [comment]
Rebased + added docs.

File: debug.js [hunk 1]
```diff
@@ -120,19 +113,24 @@ function debug(namespace) {
...
     var logFn = enabled.log || exports.log || console.log.bind(console);
     logFn.apply(self, args);
   }
-  enabled.enabled = true;
 
-  var fn = exports.enabled(namespace) ? enabled : disabled;
+  debug.namespace = namespace;
+  debug.enabled = exports.enabled(namespace);
+  debug.useColors = exports.useColors();
+  debug.color = selectColor(namespae);
```
Comment (fengmk2) [inline debug.js]: `namespae` typo here!!!

**Pull request merged**